homeassistant/components/homekit/type_security_systems.py

```diff
@@ -81,15 +83,15 @@ def update_state(self, new_state):
         hass_state = new_state.state
         if hass_state in HASS_TO_HOMEKIT:
             current_security_state = HASS_TO_HOMEKIT[hass_state]
-            self.char_current_state.set_value(current_security_state)
-            _LOGGER.debug(
-                "%s: Updated current state to %s (%d)",
-                self.entity_id,
-                hass_state,
-                current_security_state,
-            )
+            if self.char_current_state.value != current_security_state:
+                self.char_current_state.set_value(current_security_state)
+                _LOGGER.debug(
+                    "%s: Updated current state to %s (%d)",
+                    self.entity_id,
+                    hass_state,
+                    current_security_state,
+                )
 
             # SecuritySystemTargetState does not support triggered
-            if not self._flag_state and hass_state != STATE_ALARM_TRIGGERED:
+            if hass_state not in (STATE_ALARM_TRIGGERED, self.char_target_state.value):
```

Comment:
char_target_state.value contains a HASS state ?

Reply:
Thanks not right. Thanks for catching that.

Fixed

Reply:
I'm now in trouble for making the security system loud to test but its working :)